wordpress-mobile/WordPress-FluxC-Android
# Check if site supports sharing

### Description

This PR
- Adds `active_modules`, `publicize_permanently_disabled` fields to `SiteWPComRestResponse`, `SiteModel`.
- Copies [sharing support logic](https://github.com/wordpress-mobile/WordPress-iOS/blob/c2246eee0d7c3681c12698c19ab65949f2e8fd73/WordPress/Classes/Models/Blog.m#L575) from the iOS app.

### How to Test
- See that tests cover important conditions
- For manual testing, see instructions in https://github.com/wordpress-mobile/WordPress-Android/pull/14979
